expression/constant_fold.go

```diff
@@ -35,6 +36,10 @@ func FoldConstant(expr Expression) Expression {
 	return e
 }
 
+func nameConstFoldHandler(expr *ScalarFunction) (Expression, bool) {
+	return foldConstant(expr.GetArgs()[1])
```

Comment:
Why do we need this?

Reply:
`name_const` always return the second parameter directly, and the second parameter must be constant. So we can directly fold `name_const` in `constant-folding` stage, and do not left it to runtime, I think.

Reply:
During plan building stage, when converting the function from AST to scalar function, expression rewriter will check if the function can be constant-folded, so for `name_const`, it would always be folded then because its arguments are constant, hence seems we don't need to specially handle it here? am I missing anything?

Reply:
uh.. it seems that a scalar function can be folded if all its arguments can be folded. In `name_const`,  the only meaningful argument to its result is the second argument, the first argument should be excluded from this procedure. So I add the special rule to indicate that. But actually as required, the first argument of `name_const` must also be constant too, so the constant-folding works as well without the special rule.

Reply:
@eurekaka  PTAL :)

Reply:
Hmm, I still think we don't need this IMHO. @zz-jason what's your opinion?

Reply:
When it's created from `NewFunction()`, the function `name_const()` is folded. This rule seems never get a chance to be applied?

Reply:
@zz-jason hmm.. I tested here and found this rule works...
@eurekaka  ok, I'll remove this rule.

Reply:
@zz-jason  I mean, `name_const` is folded, this special rule is not applied, indeed.